[stable-2.16] Downgrade sphinx and fix 404 pages

gotmax23 opened the pull request:
Relates: https://github.com/ansible/ansible-documentation/issues/678

gotmax23 committed:
downgrade sphinx to 7.1.2 to fix 404 pages

Relates: https://github.com/ansible/ansible-documentation/issues/678

felixfontein commented:
I think the second commit should be removed from here, and instead should come to stable-2.16 as a backport of #827.

gotmax23 committed:
sphinx: remove usage of deprecated notfound-page options (#827)

sphinx-notfound-page deprecated and removed notfound_no_urls_prefix,
notfound_default_language, and notfound_default_version in favor of
notfound_urls_prefix.
We were still using these old options which caused our 404 pages to
break.

(cherry picked from commit 89ef0d9d2a52a048615a6b19104eca9499eebfdd)

gotmax23 commented:
> I think the second commit should be removed from here, and instead should come to stable-2.16 as a backport of https://github.com/ansible/ansible-documentation/pull/827.


See https://github.com/ansible/ansible-documentation/pull/827#issuecomment-1823804916. I can split them back out if you think that would make more sense.

gotmax23 commented:
@felixfontein, PTAL. I changed this to `git cherry-pick` the merge commit from https://github.com/ansible/ansible-documentation/pull/827 instead of the original commit from my PR branch.

samccann commented:
Verified on test:
http://docs.testing.ansible.com/ansible/devel/inde2x.html shows the good 404 page.
